src/Features/Core/Portable/ReplaceConditionalWithStatements/AbstractReplaceConditionalWithStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,295 @@
+            invocationExpression.Parent is TExpressionStatementSyntax expressionStatement)
+        {
+            context.RegisterRefactoring(CodeAction.Create(
+                FeaturesResources.Replace_conditional_expression_with_statements,
+                c => ReplaceConditionalExpressionInInvocationExpressionAsync(
+                    document, conditionalExpression, argument, invocationExpression, expressionStatement, c)));
+            return;
+        }
+    }
+
+    private static SyntaxNode TryCast(SyntaxGenerator generator, SyntaxNode whenTrue, ITypeSymbol? conditionalType)
+        => conditionalType is null or IErrorTypeSymbol
+            ? whenTrue
+            : generator.CastExpression(conditionalType, whenTrue);
+
+    private static TSyntaxNode WithElasticTrivia<TSyntaxNode>(SyntaxGenerator generator, TSyntaxNode node) where TSyntaxNode : SyntaxNode
+        => node.WithLeadingTrivia(generator.ElasticMarker).WithTrailingTrivia(generator.ElasticMarker);
+
+    private static async Task<Document> ReplaceConditionalExpressionInAssignmentStatementAsync(
+        Document document,
+        TConditionalExpressionSyntax conditionalExpression,
+        TStatementSyntax assignmentStatement,
+        CancellationToken cancellationToken)
+    {
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        var generator = SyntaxGenerator.GetGenerator(document);
+        var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+        // When we have `a = x ? y : z`, then the type of 'y' and 'z' can influence each other.
+        // If we convert this into:
+        //
+        //   if (x)
+        //     a = y;
+        //   else
+        //     a = z;
+        //
+        // Then we need to preserve that meaning so that 'y' and 'z' have the same type/value, even after the
+        // transformation.
+        var conditionalType = semanticModel.GetTypeInfo(conditionalExpression, cancellationToken).Type;
+
+        syntaxFacts.GetPartsOfAssignmentStatement(assignmentStatement, out var left, out _);
+        syntaxFacts.GetPartsOfConditionalExpression(conditionalExpression, out var condition, out var whenTrue, out var whenFalse);
+        var ifStatement = generator.IfStatement(
+            condition,
+            new[] { generator.AssignmentStatement(left, TryCast(generator, whenTrue, conditionalType)) },
+            new[] { generator.AssignmentStatement(left, TryCast(generator, whenFalse, conditionalType)) }).WithTriviaFrom(assignmentStatement);
+
+        var editor = new SyntaxEditor(root, generator);
+        editor.ReplaceNode(
+            assignmentStatement,
+            ifStatement);
+
+        return document.WithSyntaxRoot(editor.GetChangedRoot());
```

Comment:
Do we need an editor? Wouldn't this work?

```suggestion
        return document.WithSyntaxRoot(root.ReplaceNode(assignmentStatement, ifStatement));
```

Reply:
editor not needed.  removed.